transaction/core/src/fog_hint.rs

```diff
@@ -109,8 +109,13 @@ impl FogHint {
         ingest_server_private_key: &RistrettoPrivate,
         ciphertext: &EncryptedFogHint,
     ) -> Result<Self, CryptoBoxError> {
-        let plaintext = VersionedCryptoBox::default()
+        let (result, plaintext) = VersionedCryptoBox::default()
             .decrypt_fixed_length(ingest_server_private_key, ciphertext.as_ref())?;
+
+        if !result {
+            return Err(CryptoBoxError::MacFailed);
```

Comment:
i guess it's still being used here but we should fix this up

Reply:
actually can we just remove the non-ct decrypt version and only have one version?

Reply:
We're still using MacFailed in all the non constant time code (CryptoBox and FogHint), so removing it seems orthogonal to this PR.

Reply:
I can totally remove the non-ct decrypt from FogHint, but there will still be some MacFailed left in CryptoBox's non-ct code.

Reply:
ok, i can make a PR for that later